src/LuaTransformer.ts

```diff
@@ -4630,6 +4630,10 @@ export class LuaTransformer {
     }
 
     public transformIdentifier(identifier: ts.Identifier): tstl.Identifier {
+        if (identifier.escapedText === "globalThis") {
```

Comment:
It should be checked in `transformIdentifierExpression` (because for example in `foo.bar` `foo` is considered identifier expression and `bar` is just identifier). It also should check it with `isStandardLibraryType`, to allow reassigning it